src/main/java/com/hivemq/configuration/service/entity/TcpListener.java

```diff
@@ -93,4 +90,49 @@ public String readableName() {
         return name;
     }
 
+    public static class Builder {
+
+        private @Nullable String name;
+        private @Nullable Integer port;
+        private @Nullable String bindAddress;
+
+        public Builder() {
+        }
+
+        public Builder(final @NotNull TcpListener tcpListener) {
+            port = tcpListener.getPort();
+            bindAddress = tcpListener.getBindAddress();
+            name = tcpListener.getName();
+        }
```

Comment:
Is this used? 
I don't like having multiple constructors for a Builder class. 
I'd rather have a static method `copy()` that internally uses the builder to create a copied object.

Reply:
I'm with you that the builders can be done better. A copy method would work out when we could have static methods here. See my comment below.